boards/xiao_m0/src/lib.rs

```diff
@@ -179,7 +179,7 @@ pub fn i2c_master(
 pub type SpiPads = Pads<Sercom0, Miso, Mosi, Sclk>;
 
 /// SPI master for the labelled SPI peripheral
-pub type Spi = spi::Spi<spi::Config<SpiPads>>;
+pub type Spi = spi::Spi<spi::Config<SpiPads>, spi::Duplex>;
```

Comment:
I don't think this is needed, I think this is default.

Reply:
build fails if the parameter is not given explicitly: 
```
   Compiling xiao_m0 v0.11.0 (/home/josfemova/REPOSITORIOS/atsamd/boards/xiao_m0)
error[E0107]: this struct takes 2 generic arguments but 1 generic argument was supplied
    --> src/lib.rs:182:21
     |
182  | pub type Spi = spi::Spi<spi::Config<SpiPads>>;
     |                     ^^^ -------------------- supplied 1 generic argument
     |                     |
     |                     expected 2 generic arguments
     |
note: struct defined here, with 2 generic parameters: `C`, `A`
    --> /home/josfemova/.cargo/registry/src/github.com-1ecc6299db9ec823/atsamd-hal-0.14.0/src/sercom/v2/spi.rs:1115:12
     |
1115 | pub struct Spi<C, A>
     |            ^^^ -  -
help: add missing generic argument
     |
182  | pub type Spi = spi::Spi<spi::Config<SpiPads>, A>;
     |                                             +++
```

Reply:
ah my bad, this was a 0.13 to 0.14 change I think.

Reply:
ah my bad, this was a 0.13 to 0.14 change I think.

Reply:
One question btw, should this hal version bump imply a bsp version bump?
I just changed that one thing, but as we would be bumping `cortex-m` and `atsamd-hal` (which also implies pac upgrade) I had that question lingering in my mind

Reply:
Yeah the publicly exposed API is changing (in this case, the exact type), so a minor version bump is needed here. Depending on how the client wrote their code it may or may not be an actually breaking change though.